From 80d225fcdbc7569e3aa97146cfe0a542b9d4d8e3 Mon Sep 17 00:00:00 2001
From: Raphael Gallais-Pou <raphael.gallais-pou@foss.st.com>
Date: Mon, 31 Jul 2023 11:29:28 +0200
Subject: [PATCH 0457/1141] drm/stm: lvds: fix miscellaneous warnings

$ ./scripts/checkpatch.pl --strict -f drivers/gpu/drm/stm/lvds.c
[...]
CHECK: Please don't use multiple blank lines
+
+

$ <smatch>/smatch_scripts/kchecker drivers/gpu/drm/stm/lvds.c
[...]
  CHECK   drivers/gpu/drm/stm/lvds.c
drivers/gpu/drm/stm/lvds.c:765 lvds_config_data_mapping() error: we previously assumed 'info->bus_formats' could be null (see line 758)
drivers/gpu/drm/stm/lvds.c:872 lvds_config_mode() error: uninitialized symbol 'lvds_cdl1cr'.
drivers/gpu/drm/stm/lvds.c:873 lvds_config_mode() error: uninitialized symbol 'lvds_cdl2cr'.

Signed-off-by: Raphael Gallais-Pou <raphael.gallais-pou@foss.st.com>
Change-Id: Ic0f4fd317cc57b67bc4c51119702df1c2eff7758
---
 drivers/gpu/drm/stm/lvds.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

--- a/drivers/gpu/drm/stm/lvds.c
+++ b/drivers/gpu/drm/stm/lvds.c
@@ -755,8 +755,10 @@ static void lvds_config_data_mapping(str
 	int i;
 
 	info = &(&lvds->connector)->display_info;
-	if (!info->num_bus_formats || !info->bus_formats)
+	if (!info->num_bus_formats || !info->bus_formats) {
 		dev_warn(lvds->dev, "No LVDS bus format reported\n");
+		return;
+	}
 
 /*      Mode mirror : TODO
  *      info->bus_flags & DRM_BUS_FLAG_DATA_LSB_TO_MSB
@@ -814,6 +816,8 @@ static int lvds_config_mode(struct stm_l
 	u32 bus_flags, lvds_cr, lvds_cdl1cr, lvds_cdl2cr;
 
 	lvds_cr = 0;
+	lvds_cdl1cr = 0;
+	lvds_cdl2cr = 0;
 
 	connector = &lvds->connector;
 	if (!connector)
@@ -1206,7 +1210,6 @@ static int lvds_remove(struct platform_d
 {
 	struct stm_lvds *lvds = platform_get_drvdata(pdev);
 
-
 	lvds_pixel_clk_unregister(lvds);
 
 	/* Unregister LVDS bridge */
